dev/RadioButtons/RadioButtons.cpp

```diff
@@ -7,136 +7,634 @@
 #include "Vector.h"
 #include "RuntimeProfiler.h"
 #include "ResourceAccessor.h"
+#include "RadioButtonsTestHooks.h"
 
 RadioButtons::RadioButtons()
 {
     __RP_Marker_ClassById(RuntimeProfiler::ProfId_RadioButtons);
 
-    auto items = winrt::make<Vector<winrt::IInspectable>>();
+    m_selectionModel = winrt::SelectionModel{};
+    m_selectionModel.SingleSelect(true);
+    m_selectionModel.Source(GetItemsSource());
+    m_selectionChangedRevoker = m_selectionModel.SelectionChanged(winrt::auto_revoke, { this, &RadioButtons::OnSelectionChanged });
+
+    auto const items = winrt::make<Vector<winrt::IInspectable>>();
     SetValue(s_ItemsProperty, items);
 
     SetDefaultStyleKey(this);
+
+    // Override normal up/down/left/right behavior -- down should always go to the next item and up to the previous.
+    // left and right should be spacial but contained to the RadioButtons control. We have to attach to PreviewKeyDown
+    // because RadioButton has a key down handler for up and down that gets called before we can intercept.
+    PreviewKeyDown({ this, &RadioButtons::OnChildKeyDown });
```

Comment:
OnPreviewKeyDown. Also, You are using Child in the name of some of the handlers, which gives the impression that the event is registered on the child but it is on the Parent.

Reply:
I'm doing this because conceptually we only care about the child raising the event, but we are depending on the bubbling of the event so that we can handle it in the parent.  For instance, RadioButtons is not a tab stop so it will never get focus, only its children will get focus, thus I've named the GotFocus handler onChildGotFocus, because the first thing we do is look at the OriginalSource property to figure out which child this came from.

---
In reply to: [347424270](https://github.com/microsoft/microsoft-ui-xaml/pull/1623#discussion_r347424270) [](ancestors = 347424270)